### PR TITLE
Fix/adf 270/feature flag for advanced search event

### PR DESCRIPTION
**Pull request by boajer**
https://oat-sa.atlassian.net/browse/ADF-270

### CODE REVIEW

File: models/classes/resources/ResourceWatcher.php [hunk 1]
```diff
@@ -100,22 +94,20 @@ public function catchUpdatedResourceEvent(ResourceUpdated $event): void
...
-     */
     public function catchDeletedResourceEvent(ResourceDeleted $event): void
     {
         $searchService = $this->getServiceLocator()->get(Search::SERVICE_ID);
         try {
             $searchService->remove($event->getId());
         } catch (\Exception $e) {
             $message = $e->getMessage();
-            $this->getLogger()->error("Error delete index document for {$event->getId()} with message $message");
+            $this->getLogger()->error(
```
Comment (gabrielfs7) [inline models/classes/resources/ResourceWatcher.php:104]: Should we only remove here if advanced search is enabled too?

**Comment by gabrielfs7** [comment]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

File: models/classes/resources/ResourceWatcher.php (hunk 1 above)
Comment (boajer) [reply models/classes/resources/ResourceWatcher.php:104]: Postponed as in scope of another task

**Review by gabrielfs7 — APPROVED** [review]
LGTM

**Review by shpran — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Pull request merged**